promql/prettier/prettier_test.go

```diff
@@ -0,0 +1,966 @@
+		expected: `  metric_one[5m]`,
+	},
+	{
+		expr: `metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m]`,
+		expected: `  metric_one{
+    _a_some_very_large_label="_a_very_large_value",
+    some_very_large_label="a_very_large_value",
+  }[5m]`,
+	},
+	{
+		expr:     `metric_one[5m:1m]`,
+		expected: `  metric_one[5m:1m]`,
+	},
+	{
+		expr: `metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m:1m]`,
+		expected: `  metric_one{
+    _a_some_very_large_label="_a_very_large_value",
+    some_very_large_label="a_very_large_value",
+  }[5m:1m]`,
+	},
+	{
+		expr: `(metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m])`,
+		expected: `  (
+    metric_one{
+      _a_some_very_large_label="_a_very_large_value",
+      some_very_large_label="a_very_large_value",
+    }[5m]
+  )`,
+	},
+	{
+		expr: `((metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m]))`,
+		expected: `  (
+    (
+      metric_one{
+        _a_some_very_large_label="_a_very_large_value",
+        some_very_large_label="a_very_large_value",
+      }[5m]
+    )
+  )`,
+	},
+	{
+		expr: `(metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m:1m])`,
+		expected: `  (
+    metric_one{
+      _a_some_very_large_label="_a_very_large_value",
+      some_very_large_label="a_very_large_value",
+    }[5m:1m]
+  )`,
+	},
+	{
+		expr: `((metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m:1m]))`,
+		expected: `  (
+    (
+      metric_one{
+        _a_some_very_large_label="_a_very_large_value",
+        some_very_large_label="a_very_large_value",
+      }[5m:1m]
+    )
+  )`,
+	},
+	{
+		expr: `((((metric_one{_a_some_very_large_label="_a_very_large_value", some_very_large_label="a_very_large_value"}[5m:1m]))))`,
+		expected: `  (
+    (
+      (
+        (
+          metric_one{
+            _a_some_very_large_label="_a_very_large_value",
+            some_very_large_label="a_very_large_value",
+          }[5m:1m]
+        )
+      )
+    )
+  )`,
+	},
+	{
+		expr:     `rate(metric_name[5m])`,
+		expected: `  rate(metric_name[5m])`,
+	},
+	{
+		expr: `rate(metric_three{a_some_very_large_label="a_very_large_value", some_very_large_label="a_very_large_value"}[5m])`,
+		expected: `  rate(
+    metric_three{
+      a_some_very_large_label="a_very_large_value",
+      some_very_large_label="a_very_large_value",
+    }[5m]
+  )`,
+	},
+	{
+		expr: `histogram_quantile(0.9, rate(http_request_duration_seconds_bucket[10m]))`,
+		expected: `  histogram_quantile(
+    0.9,
+    rate(http_request_duration_seconds_bucket[10m])
+  )`,
+	},
+	{
+		expr: `label_join(up{job="api-server",src1="a",src2="b",src3="c"}, "foo", ",", "src1", "src2", "src3")`,
+		expected: `  label_join(
```

Comment:
This shouldn't be indented either.

Reply:
Addressed in recent commits. Please resolve those comments that are satisfactory.

Reply:
This is kept since while printing in block scalar format, we are adding 2 spaces to give a better look.

Reply:
Also, this has a depth of 1, so `2 spaces` in the AST, keeping things simple.

Reply:
I'm not following at all, there's no need for indenting here. If a caller wants to indent more, that's up to them.

Reply:
Please resolve if it is satisfactory.